tools/src/scie_pants/update_scie_pants.py

```diff
@@ -0,0 +1,288 @@
+    )
+
+    latest_releases = []
+    for release_data in releases:
+        if release_data.get("draft") or release_data.get("prerelease"):
+            continue
+        tag_name = release_data.get("tag_name")
+        if not tag_name:
+            continue
+        match = RELEASE_TAG_MATCHER.match(tag_name)
+        if not match:
+            log.debug(
+                f"Skipping tag {tag_name} since it does not match {RELEASE_TAG_MATCHER.pattern}"
+            )
+            continue
+        version = Version(match["version"])
+        release = Release.from_api_response(version, platform, release_data)
+        if release:
+            latest_releases.append(release)
+
+    if not latest_releases:
+        log.debug(
+            f"No releases for {BINARY_NAME} compatible with {platform} found in: "
+            f"{json.dumps(releases, indent=2)}"
+        )
+        return None
+
+    # We want the highest version; so this needs to be a reverse sort.
+    latest_releases.sort(reverse=True, key=lambda rel: rel.version)
+    return latest_releases[0]
+
+
+def install_release(ptex: Ptex, release: Release, scie: Path) -> Path:
+    # The `.sha256` checksum file format is a single line with two fields, space separated.
+    # The 1st field is the hexadecimal checksum and the second field the name of the file it applies
+    # to. See: https://man7.org/linux/man-pages/man1/sha256sum.1.html
+    expected_sha256, _ = ptex.fetch_text(release.binary_sha256_url).strip().split(" ", maxsplit=1)
+
+    download_dir = Path(tempfile.mkdtemp())
+    atexit.register(shutil.rmtree, download_dir, ignore_errors=True)
+
+    binary = download_dir / release.file_name
+    with open(binary, "wb") as fp:
+        ptex.fetch_to_fp(release.binary_url, fp)
+
+    # TODO(John Sirois): Ideally this would not be 2-pass and we'd hash the download stream inline.
+    digest = hashlib.sha256()
+    with open(binary, "rb") as fp:
+        for chunk in iter(lambda: fp.read(io.DEFAULT_BUFFER_SIZE), b""):
+            digest.update(chunk)
+    actual_sha256 = digest.hexdigest()
+    if expected_sha256 != actual_sha256:
+        eol = os.linesep
+        raise ValueError(
+            f"The binary downloaded from {release.binary_url} is invalid.{eol}"
+            f"The expected fingerprint from {release.binary_sha256_url} was:{eol}"
+            f"  {expected_sha256}{eol}"
+            f"The actual fingerprint of the downloaded file is:{eol}"
+            f"  {actual_sha256}",
+        )
+
+    # Mark the binary as executable. This is needed on Unix but not on Windows, where its harmless.
+    binary.chmod(0o755)
+
+    # We first move the running scie before replacing it. This satisfies Windows constraints
+    # surrounding manipulation of running binaries.
+    backup = scie.rename(scie.with_suffix(".bak"))
+    try:
+        shutil.move(str(binary), str(scie))
+    except OSError:
+        warn(f"A backup is saved in {backup}")
```

Comment:
Why not try moving the backup back here?

Reply:
I think once you start to run into issues its best to hands off and give clear info. If a _second_ error happened trying to do too much ... that's just a mess. Right now you get a clear message as to where the binary was moved to. It's very easy to just cut/paste move it back.

Reply:
This will definitely require some playing around with. To even arrive at the move first then replace required a bunch of messing around testing on Windows, which is what this is all dealing with. Simple on Unix.